### PR TITLE
Changes from my side

### PR DESCRIPTION
**Pull request by oleksiyp**
Latest MockK version
clearAllMocks() instead of clearMocks()
confirmVerified with exclusion
Execute all tests using JUnit platform

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -29,6 +29,8 @@ android {
     }
 
     testOptions.unitTests.all {
+        useJUnitPlatform()
```
Comment (oleksiyp) [inline app/build.gradle:32]: Allow executing both old JUnit 4 and JUnit 5 tests in one run

File: app/src/test/java/com/wiseassblog/kotlincalculator/CalculatorPresenterTest.kt [hunk 2]
```diff
@@ -59,6 +59,10 @@ class CalculatorPresenterTest {
         //These are the assertions which must be satisfied in order to pass the test
         coVerify { eval.evaluateExpression(EXPRESSION) }
         coVerify { viewModel.setDisplayState(ANSWER) }
+
+        // but as well some not so important calls
+        excludeRecords { viewModel.setObserver(any()) }
+        confirmVerified(eval, viewModel)
```
Comment (oleksiyp) [inline app/src/test/java/com/wiseassblog/kotlincalculator/CalculatorPresenterTest.kt:65]: Makes final check that all recorded calls are covered with verification. Similar to `verifyNoMoreInteractions` in Mockito.

Comment (oleksiyp) [reply app/src/test/java/com/wiseassblog/kotlincalculator/CalculatorPresenterTest.kt:65]: `excludeRecords` needed to skip not important for verification calls

File: versions.gradle [hunk 3]
```diff
@@ -37,8 +37,10 @@ deps.android = android
 def test = [:]
 
 test.junit = "org.junit.jupiter:junit-jupiter-api:$versions.junit"
+test.jupiter_engine = "org.junit.jupiter:junit-jupiter-engine:$versions.junit"
+test.vintage_engine = "org.junit.vintage:junit-vintage-engine:$versions.junit"
 test.kotlin_junit = "org.jetbrains.kotlin:kotlin-test-junit:$versions.kotlin"
-test.mockk = "io.mockk:mockk:1.8.9.kotlin13"
+test.mockk = "io.mockk:mockk:1.9"
```
Comment (oleksiyp) [inline versions.gradle:43]: Latest release (needed for `confirmVerified`)

**Comment by oleksiyp** [comment]
Overall only one thing is wrong: `clearMocks` without arguments. It should be replaced with `clearAllMocks()` or `clearMocks(mock1)`. All the rest is just suggestions

**Comment by oleksiyp** [comment]
Opened https://github.com/mockk/mockk/issues/217 to address `clearMocks` usability issue

File: app/src/test/java/com/wiseassblog/kotlincalculator/CalculatorPresenterTest.kt [hunk 4]
```diff
@@ -38,7 +38,7 @@ class CalculatorPresenterTest {
 
     @BeforeEach
     fun setUpRedundantMocks() {
-        clearMocks()
+        clearAllMocks()
```
Comment (oleksiyp) [inline app/src/test/java/com/wiseassblog/kotlincalculator/CalculatorPresenterTest.kt:41]: Critical thing, `clearMocks` require as input mocks to be cleared.

**Comment by BracketCove** [comment]
Looks great, thank you for the changes and info @oleksiyp!

**Pull request merged**